src/AmqpConnectionManager.rei

```diff
@@ -4,27 +4,23 @@ type urls = array(url);
 exception ConnectionError(Js.Exn.t);
 
 module Queue: {
-  type name;
+  type name = string;
   type message = {content: Node.Buffer.t};
 };
 
-module Exchange: {type name;};
+module Exchange: {type name = string;};
 
 module Channel: {
   type t;
-  type name;
+  type name = string;
   type ack = Queue.message => unit;
   type nack = Queue.message => unit;
 
   let ack: t => ack;
   let nack: t => nack;
 
   module Config: {
-    type nonrec t = {
-      .
-      "json": bool,
-      "setup": t => Js.Promise.t(unit),
-    };
+    type nonrec t('a) = {.. "setup": t => Js.Promise.t(unit)} as 'a;
```

Comment:
I'm pretty sure this is bad practice, but I'm not sure of a better way to do this.

I'd like to ensure that `"setup"` is of the specified type, but I don't really care about any properties or their types beyond that.